typespec-extension/src/code-model-builder.ts

```diff
@@ -246,7 +237,9 @@ export class CodeModelBuilder {
   }
 
   public async build(): Promise<CodeModel> {
-    this.sdkContext = await createSdkContext(this.emitterContext, "@azure-tools/typespec-java");
+    this.sdkContext = await createSdkContext(this.emitterContext, "@azure-tools/typespec-java", {
+      versioning: { previewStringRegex: /$/ },
```

Comment:
Does it need to depends on the `options["service-version-exclude-preview"]`?

Reply:
Yes, the current logic is to get all the api-versions and do the filtering by ourselves. This is the logics to get `options["service-version-exclude-preview"]` and filter the api-versions: https://github.com/Azure/autorest.java/blob/c007e8225c52dfd4c53446fd7ea0785ddeee8d74/typespec-extension/src/code-model-builder.ts#L518-L521

Reply:
Should we migrate to use TCGC on this?

Reply:
I'm not sure. I may tend to keep the logics for now as using the regex to keep preview is not an ideal way. I will discuss with Chenjie and Crystal on how TCGC support filtering api versions, I will update the code after TCGC design finalized. I can create an issue to track.

Reply:
OK

Reply:
issue created： https://github.com/Azure/autorest.java/issues/2913